src/app/shared/components/template/components/layout/nav_group.ts

```diff
@@ -86,8 +87,30 @@ export class NavGroupComponent extends TemplateLayoutComponent {
   }
 
   modifyRowSetter(row: FlowTypes.TemplateRow) {
-    if (Array.isArray(row?.value)) {
-      this.templateNames = row.value;
+    // Check if value is an object referring to a whole data_list with a "template" column,
+    // and if so, assign an array of the values of this column to templateNames.
+    // Handle case where object is passed in directly, or where inside a "_list" local variable
+    const templateArray = [];
+    const dataListObject = isObject(row?.value)
+      ? row.value
+      : Array.isArray(row?.value) && isObject(row?.value[0])
+      ? row.value[0]
+      : null;
```

Comment:
I would prefer these last 2 lines to be indented for clarity, but the indent gets removed by the linter on commit